### PR TITLE
xet cp source accepts list, supporting real *

### PR DESCRIPTION
**Pull request by seanses**
Fix https://github.com/xetdata/xethub/issues/3868
Screenshot 2023-11-21 at 11 51 21 AM

### CODE REVIEW

File: python/pyxet/pyxet/util.py [hunk 1]
```diff
@@ -49,14 +49,32 @@ def _get_fs_and_path(uri, strip_trailing_slash = True):
...
 def _isdir(fs, path):
     if fs.protocol == 'xet':
         return fs.isdir_or_branch(path)
     else:
         return fs.isdir(path)
 
+def _get_fs_string(uri):
+    if uri.find('://') == -1:
+        return "local"
+    split = uri.split("://")
```
Comment (hoytak) [inline python/pyxet/pyxet/util.py:71]: Can just do:
n = uri.find('://')
if n == -1:
    return "local"
else:
   return uri[:n]

**Review by hoytak — APPROVED** [review]
Small updates; but looks good to me!

File: python/pyxet/pyxet/cli.py [hunk 2]
```diff
@@ -125,7 +125,7 @@ def cp(source: Annotated[str, typer.Argument(help="Source file or folder to copy
                int, typer.Option("--parallel", "-p", help="Maximum amount of parallelism")] = 32):
         """copy files and folders"""
         if not message:
-            message = f"copy {source} to {target}" if not recursive else f"copy {source} to {target} recursively"
+            message = f"copy {source[0]}... to {target}" if not recursive else f"copy {source[0]}... to {target} recursively"
```
Comment (hoytak) [inline python/pyxet/pyxet/cli.py:128]: Will crash on empty list...  
Let's actually do `{source[:3].join(', ')}...`

Comment (seanses) [reply python/pyxet/pyxet/cli.py:128]: Added a check for empty source list.

**Pull request merged**